File: codegraph/testdata/tsproj/service.ts
export interface Runner {
  run(): string;
}

export class Base {
  protected id = 0;
}

export class Service extends Base implements Runner {
  constructor(private name: string) {
    super();
  }

  run(): string {
    return this.name + ': running';
  }

  stop(): void {
    cleanup();
  }
}

function cleanup(): void {}

File: codegraph/testdata/pyproj/service.py
class Base:
    def __init__(self):
        self.id = 0


class Service(Base):
    def __init__(self, name):
        super().__init__()
        self.name = name

    def run(self):
        return self.name + ": running"

    def stop(self):
        cleanup()


def cleanup():
    pass
